prime_sieve.py

```
@@ -0,0 +1,66 @@
+        return []
+    sieve = bytearray(b"\x01") * (limit + 1)
+    sieve[:2] = b"\x00\x00"  # 0 and 1 are not prime
+    for num in range(2, isqrt(limit) + 1):
+        if sieve[num]:
+            step = num
+            start = num * num
+            sieve[start: limit + 1: step] = b"\x00" * ((limit - start) // step + 1)
+    return [i for i, is_prime in enumerate(sieve) if is_prime]
+
+
+def segmented_sieve(limit, segment_size=32768):
```

Ruff (0.11.9)

Missing return type annotation for public function `segmented_sieve`

(ANN201)